lib/features/modeling/behavior/SubProcessStartEventBehavior.js

```diff
@@ -0,0 +1,80 @@
+import inherits from 'inherits';
+
+import CommandInterceptor from 'diagram-js/lib/command/CommandInterceptor';
+
+import { is } from '../../../util/ModelUtil';
+import { isExpanded } from '../../../util/DiUtil.js';
+
+/**
+ * Add start event child by default when creating an expanded subprocess
+ * with create.start or replacing a task with an expanded subprocess.
+ */
+export default function SubProcessStartEventBehavior(eventBus, modeling) {
+  CommandInterceptor.call(this, eventBus);
+
+  eventBus.on('create.start', function(event) {
+    var shape = event.context.shape,
+        hints = event.context.hints;
+
+    if (
+      !is(shape, 'bpmn:SubProcess') &&
+      isExpanded(shape)
+    ) {
+      return;
+    }
+
+    hints.shouldAddStartEvent = true;
+  });
+
+  this.postExecuted('shape.create', function(event) {
+    var shape = event.context.shape,
+        hints = event.context.hints,
+        position;
+
+    if (
```

Comment:
Why do we need to double check here rather than simply checking `hints.shouldAddStartEvent`?

Reply:
You're right, just checking `hints.shouldAddStartEvent` is sufficient here.

I was still thinking of the other cases where `shape.create` is invoked but that won't be an issue with the new hints flag since it will be undefined in those situations.